### PR TITLE
1436790/1438462: Part 6: 30 files with MSDN link fixes.

### PR DESCRIPTION
**Pull request by DennisLee-DennisLee**
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

### CODE REVIEW

File: docs/visual-basic/programming-guide/concepts/collections.md [hunk 1]
```diff
@@ -456,7 +456,7 @@ End Class
   
 <a name="BKMK_CustomCollection"></a> 
 ## Defining a Custom Collection  
- You can define a collection by implementing the <xref:System.Collections.Generic.IEnumerable%601> or <xref:System.Collections.IEnumerable> interface. For additional information, see [Enumerating a Collection](https://msdn.microsoft.com/library/71807ea7-9180-48a6-916f-35a5251d477f).  
+ You can define a collection by implementing the <xref:System.Collections.Generic.IEnumerable%601> or <xref:System.Collections.IEnumerable> interface. For additional information, see [Enumerating a Collection](https://docs.microsoft.com/previous-versions/hwyysy67(v=vs.140)).
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/concepts/collections.md]:
```suggestion
 You can define a collection by implementing the <xref:System.Collections.Generic.IEnumerable%601> or <xref:System.Collections.IEnumerable> interface. For additional information, see [Enumerating a Collection](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/hwyysy67(v=vs.100)).
```

File: docs/visual-basic/programming-guide/language-features/objects-and-classes/object-lifetime-how-objects-are-created-and-destroyed.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ End Sub
   
 ## See also
 - <xref:System.IDisposable.Dispose%2A>
-- [Initialization and Termination of Components](https://msdn.microsoft.com/library/58444076-a9d2-4c91-b3f6-0e180dc0695d)
+- [Initialization and Termination of Components](https://docs.microsoft.com/previous-versions/ws9dc6t6(v=vs.140))
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/language-features/objects-and-classes/object-lifetime-how-objects-are-created-and-destroyed.md]:
```suggestion
- [Initialization and Termination of Components](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2013/ws9dc6t6(v=vs.120))
```

File: docs/visual-basic/programming-guide/language-features/operators-and-expressions/index.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ An *operator* is a code element that performs an operation on one or more code e
   
  In the preceding example, Visual Basic performs the operations in the expression on the right side of the assignment operator (`=`), then assigns the resulting value to the variable `x` on the left. There is no practical limit to the number of operators that can be combined into an expression, but an understanding of [Operator Precedence in Visual Basic](../../../../visual-basic/language-reference/operators/operator-precedence.md) is necessary to ensure that you get the results you expect.  
   
- For more information and examples, see [Operator Overloading in Visual Basic 2005](https://msdn.microsoft.com/library/ms379613(v=vs.80).aspx).  
+ For more information and examples, see [Operator Overloading in Visual Basic 2005](https://docs.microsoft.com/previous-versions/ms379613(v=vs.80)).
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/language-features/operators-and-expressions/index.md]: remove this sentence

```suggestion
```

File: docs/visual-basic/programming-guide/language-features/procedures/how-to-define-an-operator.md [hunk 4]
```diff
@@ -30,7 +30,7 @@ If you have defined a class or structure, you can define the behavior of a stand
   
  [!code-vb[VbVbcnProcedures#26](./codesnippet/VisualBasic/how-to-define-an-operator_2.vb)]  
   
- For more information and examples, see [Operator Overloading in Visual Basic 2005](https://msdn.microsoft.com/library/ms379613(v=vs.80).aspx).  
+ For more information and examples, see [Operator Overloading in Visual Basic 2005](https://docs.microsoft.com/previous-versions/ms379613(v=vs.80)).
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/language-features/procedures/how-to-define-an-operator.md]:
```suggestion
```

File: docs/visual-basic/programming-guide/language-features/procedures/operator-procedures.md [hunk 5]
```diff
@@ -73,7 +73,7 @@ An operator procedure is a series of Visual Basic statements that define the beh
   
  [!code-vb[VbVbcnProcedures#24](./codesnippet/VisualBasic/operator-procedures_2.vb)]  
   
- For more information and examples, see [Operator Overloading in Visual Basic 2005](https://msdn.microsoft.com/library/ms379613(v=vs.80).aspx).  
+ For more information and examples, see [Operator Overloading in Visual Basic 2005](https://docs.microsoft.com/previous-versions/ms379613(v=vs.80)).
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/language-features/procedures/operator-procedures.md]:
```suggestion
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Looks good @DennisLee-DennisLee. Left some comments for you though. Thanks!

File: docs/visual-basic/misc/source-name-specified-in-eventlogsource-is-registered-to-another-log.md [hunk 6]
```diff
@@ -13,7 +13,7 @@ The `EventLog` is attempting to refer to a source that is registered to a differ
 -   Make sure the source is registered to the correct log. To do this, use the <xref:System.Diagnostics.EventLog.CreateEventSource%2A> method or one of its overloads to specify a string that uniquely identifies your component to the event log.  
   
 ## See also
-- [Administering Event Logs](https://msdn.microsoft.com/library/35f53238-bdd2-417b-acd8-2fd9f7397f18)
-- [Event Log References](https://msdn.microsoft.com/library/4af0661c-6c96-49f4-961d-b26ed9bc3e87)
-- [How to: Add Your Application as a Source of Event Log Entries](https://msdn.microsoft.com/library/948ff920-a739-4e66-a191-ee951512d42c)
-- [How to: Remove an Event Source](https://msdn.microsoft.com/library/bc66c900-4b8a-426a-b8e2-17031a20167e)
+- [Administering Event Logs](https://docs.microsoft.com/previous-versions/4f69axw4(v=vs.100))
```
[the remark this answers is not in the document]
Comment (DennisLee-DennisLee) [reply docs/visual-basic/misc/source-name-specified-in-eventlogsource-is-registered-to-another-log.md]: Sorry, the trick had been working for me before, so I was curious why it was failing so often in this PR. I don't have a sense of how often that trick should be successful. (To be sure, we don't need to bother with version numbers higher than what the redirect shows?)

I think I forgot to add the period between the "vs" and the version number here. That's probably why the trick wasn't working for me.

**Comment by DennisLee-DennisLee** [comment]
#sign-off

**Review by mairaw — APPROVED** [review]
Thank you @DennisLee-DennisLee! 🚀

**Pull request merged**